### PR TITLE
Include the interface name in the error message

### PR DESCRIPTION
**Pull request by manuelbuil**
#### Proposed Changes ####

Add the name of the interface to the error messages to help debugging

#### Types of Changes ####

Bugfix

#### Verification ####

Use k3s rootless as querying the interface will fail

#### Linked Issues ####

https://github.com/k3s-io/k3s/issues/8345

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

**Review by osodracnai — APPROVED** [review]

**Review by dereknola — APPROVED** [review]

File: pkg/agent/config/config.go [hunk 1]
```diff
@@ -363,7 +363,7 @@ func get(ctx context.Context, envInfo *cmds.Agent, proxy proxy.Proxy) (*config.N
 	if controlConfig.FlannelBackend != config.FlannelBackendNone && len(envInfo.FlannelIface) > 0 {
 		flannelIface, err = net.InterfaceByName(envInfo.FlannelIface)
 		if err != nil {
-			return nil, errors.Wrapf(err, "unable to find interface")
+			return nil, errors.Wrapf(err, "unable to find interface: %s", envInfo.FlannelIface)
```
Comment (brandond) [inline pkg/agent/config/config.go:366]:
```suggestion
			return nil, errors.Wrapf(err, "unable to find interface %s", envInfo.FlannelIface)
```

File: pkg/util/net.go [hunk 2]
```diff
@@ -301,14 +301,13 @@ func IPStringToIPNet(address string) (*net.IPNet, error) {
...
 
 // GetIPFromInterface is the public function that returns the IP of an interface
-func GetIPFromInterface(ifaceName string) string {
+func GetIPFromInterface(ifaceName string) (string, error) {
 	ip, err := getIPFromInterface(ifaceName)
 	if err != nil {
-		logrus.Warn(fmt.Errorf("unable to get global unicast ip from interface name: %w", err))
-	} else {
-		logrus.Infof("Found ip %s from iface %s", ip, ifaceName)
+		return "", fmt.Errorf("interface: %s does not have a correct global unicast ip: %w", ifaceName, err)
```
Comment (brandond) [inline pkg/util/net.go:307]:
```suggestion
		return "", fmt.Errorf("interface %s does not have a valid global unicast ip: %w", ifaceName, err)
```

**Review by brandond — CHANGES_REQUESTED** [review]
`:` is usually used to delimit separate messages in the error stack, we should avoid using it as part of the message.

File: pkg/agent/config/config.go (hunk 1 above)
Comment (manuelbuil) [reply pkg/agent/config/config.go:366]: ok, thanks!

**Review by brandond — APPROVED** [review]

**Review by brandond — APPROVED** [review]

**Pull request merged**